src/decorator/metadata/metadata.ts

```diff
@@ -72,6 +72,7 @@ export class Metadata<T> {
    * @throws Throws an error if an indexName was delivered but no index was found for given name or the found index has no sort key defined
    */
   getSortKey(indexName?: string): keyof T | null {
+    // todo: is this ok?! it returns null when no sortKey is defined but throws when no sortKey when index was given
```

Comment:
good question, I don't want to change it know, internally we never use the getSortKey() with indexName, so we only use the null version, it is not perfect, but would leave it for now